pkg/component/observability/opentelemetry/operator/opentelemetryoperator_test.go

```diff
@@ -0,0 +1,506 @@
+						{
+							ContainerName: vpaautoscalingv1.DefaultContainerResourcePolicy,
+							MinAllowed: corev1.ResourceList{
+								corev1.ResourceMemory: resource.MustParse("128Mi"),
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		operatorManagedResource = &resourcesv1alpha1.ManagedResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      OperatorManagedResourceName,
+				Namespace: namespace,
+			},
+		}
+		operatorManagedResourceSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "managedresource-" + operatorManagedResource.Name,
+				Namespace: namespace,
+			},
+		}
+	})
+
+	Describe("#Deploy", func() {
+		It("should successfully deploy all resources", func() {
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operatorManagedResource), operatorManagedResource)).To(BeNotFoundError())
+
+			Expect(component.Deploy(ctx)).To(Succeed())
+
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operatorManagedResource), operatorManagedResource)).To(Succeed())
+			expectedMr := &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            OperatorManagedResourceName,
+					Namespace:       namespace,
+					Labels:          map[string]string{v1beta1constants.GardenRole: "seed-system-component"},
+					ResourceVersion: "1",
+				},
+				Spec: resourcesv1alpha1.ManagedResourceSpec{
+					Class: ptr.To("seed"),
+					SecretRefs: []corev1.LocalObjectReference{{
+						Name: operatorManagedResource.Spec.SecretRefs[0].Name,
+					}},
+					KeepObjects: ptr.To(false),
+				},
+			}
+			utilruntime.Must(references.InjectAnnotations(expectedMr))
+			Expect(operatorManagedResource).To(DeepEqual(expectedMr))
+			Expect(operatorManagedResource).To(consistOf(
+				serviceAccount,
+				clusterRole,
+				clusterRoleBinding,
+				role,
+				roleBinding,
+				deployment,
+				vpa,
+			))
+
+			operatorManagedResourceSecret.Name = operatorManagedResource.Spec.SecretRefs[0].Name
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operatorManagedResourceSecret), operatorManagedResourceSecret)).To(Succeed())
+			Expect(operatorManagedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+			Expect(operatorManagedResourceSecret.Immutable).To(Equal(ptr.To(true)))
+			Expect(operatorManagedResourceSecret.Labels["resources.gardener.cloud/garbage-collectable-reference"]).To(Equal("true"))
+		})
+	})
+
+	Describe("#Destroy", func() {
+		It("should successfully destroy all resources", func() {
+			Expect(c.Create(ctx, operatorManagedResource)).To(Succeed())
+			Expect(c.Create(ctx, operatorManagedResourceSecret)).To(Succeed())
+
+			Expect(component.Destroy(ctx)).To(Succeed())
+
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operatorManagedResource), operatorManagedResource)).To(BeNotFoundError())
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operatorManagedResourceSecret), operatorManagedResourceSecret)).To(BeNotFoundError())
```

Comment:
I would explicitly try to get these resources before calling `Destroy` just to ensure they indeed exist.